tests/shell.test.ts

```diff
@@ -0,0 +1,252 @@
+import { describe, it, expect } from 'vitest';
+import * as fish from '../src/fish';
+import * as bash from '../src/bash';
+import * as zsh from '../src/zsh';
+import * as powershell from '../src/powershell';
+import { ShellCompDirective } from '../src';
+
+describe('shell completion generators', () => {
+  const name = 'testcli';
+  const exec = '/usr/bin/node /path/to/testcli';
+  const specialName = 'test-cli:app';
+  const escapedName = specialName.replace(/[-:]/g, '_');
+
+  describe('fish shell completion', () => {
+    it('should generate a valid fish completion script', () => {
+      const script = fish.generate(name, exec);
+
+      // Check that the script contains the shell name
+      expect(script).toContain(`# fish completion for ${name}`);
```

Comment:
here you can probably use `expect(script).toMatchSnapshot()` and drop all of these individual `expect` assertions

then just review the snapshot it generates